eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/ResourceClientProvider.cs

```diff
@@ -225,6 +232,78 @@ protected MethodProvider BuildValidateResourceIdMethod()
             return new MethodProvider(signature, bodyStatements, this);
         }
 
+        private CSharpType GetPathParameterType(string parameterName)
+        {
+            var operation = InputOperationForRequestPath;
+
+            var operationParameter = operation.Parameters.First(p => p.Name.Equals(parameterName, StringComparison.OrdinalIgnoreCase));
+
+            var csharpType = ManagementClientGenerator.Instance.TypeFactory.CreateCSharpType(operationParameter.Type) ?? throw new InvalidOperationException($"Could not create C# type for parameter '{parameterName}' with type '{operationParameter.Type}' in operation '{operation.Name}'.");
+            return parameterName switch
+            {
+                "subscriptionId" when csharpType.Equals(typeof(Guid)) => typeof(string),
+                "location" when csharpType.Equals(typeof(string)) => typeof(AzureLocation),
```

Comment:
I can understand subscriptionId type not consistent between different versions of common-types.
why is location needed? should we fix the spec instead?